### PR TITLE
Expose RgbWallet without Electrum dependency

### PR DESCRIPTION
**Pull request by crisdut**
The goal is to allow use RgbWallet with alternative Blockchain explorer implementation.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -27,16 +27,22 @@ extern crate log;
...
+#[cfg(feature = "electrum")]
+mod electrum;
+#[cfg(feature = "cli")]
 mod runtime;
 mod wallet;
 
 pub mod prelude {
     pub use descriptor::{RgbDescr, SpkDescriptor, Tapret, TerminalPath};
+    #[cfg(feature = "electrum")]
+    pub use electrum::BlockchainResolver;
```
Comment (dr-orlovsky) [inline src/lib.rs:39]: I think we should remove this from the prelude

File: src/runtime.rs [hunk 2]
```diff
@@ -36,8 +36,7 @@ use rgbstd::{validation, Chain};
 use strict_types::encoding::{DeserializeError, Ident, SerializeError};
 
 use crate::descriptor::RgbDescr;
-use crate::wallet::BlockchainResolver;
-use crate::{RgbWallet, Tapret};
+use crate::{BlockchainResolver, RgbWallet, Tapret};
```
Comment (dr-orlovsky) [inline src/runtime.rs:39]: BlockchainResolver must be behind feature gate

File: src/wallet.rs [hunk 3]
```diff
@@ -90,116 +88,14 @@ pub trait DefaultResolver {
...
-                ),
-                derivation,
-                amount: res.value,
+impl DefaultResolver for Chain {
+    fn default_resolver(&self) -> String {
+        match self {
+            Chain::Bitcoin => s!("blockstream.info:110"),
+            Chain::Testnet3 => s!("blockstream.info:143"),
+            Chain::Regtest => s!("localhost:50001"),
+            chain => {
```
Comment (dr-orlovsky) [inline src/wallet.rs:97]: This is also electrum-specific so should be moved into the electrum module

File: src/lib.rs [hunk 4]
```diff
@@ -27,16 +27,22 @@ extern crate log;
 extern crate serde;
 
 mod descriptor;
+#[cfg(feature = "electrum")]
+mod electrum;
+#[cfg(feature = "cli")]
```
Comment (dr-orlovsky) [inline src/lib.rs:32]: Runtime is used outside of cli, for instance in WASM. So we do not need this feature gate.

**Review by dr-orlovsky — CHANGES_REQUESTED** [review]
I think this is correct concept and approach, ACK. However some changes are needed (see my comments)

File: src/lib.rs (hunk 4 above)
Comment (crisdut) [reply src/lib.rs:32]: Ok, 

I will need change `Runtime`, because `BlockchainExplorer` had **ElectrumClient** dependecy.

Comment (dr-orlovsky) [reply src/lib.rs:32]: I think we need to refactor `BlockchainExplorer` type into an enum, where ElectrumClient is just one of the options.

The whole point of this crate is to provide `Runtime`. The rest is done by `rgb-wallet` and `rgb-std`. Basically, `rgb` crate is just `rgb-wallet` crate + FS storage + electrum.

**Comment by dr-orlovsky** [comment]
Seems this needs rebase

**Pull request closed without merging**